app/models/service_template.rb

```diff
@@ -45,6 +45,19 @@ class ServiceTemplate < ApplicationRecord
   virtual_has_one :custom_actions, :class_name => "Hash"
   virtual_has_one :custom_action_buttons, :class_name => "Array"
 
+  # This method creates
+  def self.create_catalog_item(options, auth_user)
+    create(options.except(:config_info)).tap do |service_template|
+      config_info = options[:config_info].except(:ae_endpoints)
+      workflow_class = MiqProvisionWorkflow.class_for_source(config_info[:src_vm_id])
+      raise 'Could not find Provision Workflow class for source VM' unless workflow_class
```

Comment:
Should allow to create a generic service template without workflow but possibly a subtype